concepts.tex

```diff
@@ -803,6 +564,321 @@
 \end{itemize}
 \end{itemdescr}
 
+\rSec1[concepts.lib.object]{Object concepts}
+
+\pnum
+This section \oldtxt{describes the foundational}\newtxt{defines} concepts that describe the basis of the
+value-oriented programming style on which the library is based. \oldtxt{The purpose of these concepts is
+to establish a foundation for equational reasoning in programs.} \ednote{These concepts reuse many of the
+names of concepts that traditionally describe features of types to describe families of object types (See
+the rationale in Appendix~\ref{decomposition}).}
+
+\rSec2[concepts.lib.object.destructible]{Concept Destructible}
+\ednote{Remove table [destructible] in [utility.arg.requirements]. Replace
+references to [destructible] with references to
+[concepts.lib.\oldtxt{corelang}\newtxt{object}.destructible].}
+
+\pnum
+\newtxt{The \tcode{Destructible} concept is the base of the hierarchy of object concepts.
+It describes properties that all such object types have in common.}
+
+\indexlibrary{\idxcode{Destructible}}%
+\begin{itemdecl}
+template <class T>
+concept bool Destructible@\newtxt{() \{}\oldtxt{ =}@
+  @\newtxt{return}@ requires (T t@\newtxt{, T* p}@) {
+    { t.@$\sim$@T() } noexcept;
+    @\newtxt{\{ \&t \} -> Same<T*>;}@
+    @\newtxt{delete p;}@
+    @\newtxt{delete[] p;}@
+  };
+@\newtxt{\}}@
+\end{itemdecl}
```

Comment:
Open question: since Constructible permits reference types, should Destructible also? Rationale in favor: (1) it's ok to let a local reference pass out of scope, and (2) `is_destructible` returns true for reference types.